jwst/photom/photom_step.py

```diff
@@ -41,6 +41,14 @@ def process(self, input):
                              "CubeModel, ImageModel, IFUImageModel, "
                              "MultiSlitModel, or MultiSpecModel.")
 
+        # check if extract_1d_step told photometry to be skipped
+        # e.g. for NIRISS SOSS data in FULL subarray
+        if input_model.meta.cal_step.photom == 'SKIPPED':
+            self.log.error('PhotomStep was set to SKIPPED by'
+                           ' a previous step. Photom will be skipped.')
+            result = input_model.copy()
+            return result
```

Comment:
I would suggest removing this and instead wrapping the final block of photom_step.py code (e.g. L68:74 in main) with a try except block, which would catch an `except {New_Photom_ModelType_Error}`, a custom error return that you could insert in photom.py L417, checking self.input for the expected model type with isinstance, e.g. a MultiSpecModel from extract1d rather than whatever model is fed to photom when extract_1d is skipped upstream.